operators/pkg/controller/elasticsearch/driver/driver.go

```diff
@@ -45,8 +46,9 @@ type Options struct {
 	// Version is the version of Elasticsearch we want to reconcile towards
 	Version version.Version
 	// Client is used to access the Kubernetes API
-	Client k8s.Client
-	Scheme *runtime.Scheme
+	Client   k8s.Client
+	Scheme   *runtime.Scheme
+	Recorder record.EventRecorder
```

Comment:
There is a `reconcileState.Recorder *events.Recorder` already available that could be reused but Kibana and APM uses a `record.EventRecorder`.
I'm not sure what to do: 
- refactor the ES controller to use only an `*events.Recorder`
- refactor the APM/Kibana controllers to use an `record.EventRecorder`
- accept the mix in the ES controller - this one sounds really wrong

Reply:
In fact, the `ReconcileElasticsearch` has already a field `recorder record.EventRecorder` but it's not propagated in the driver like it's done in Kibana.

Reply:
I think the original intention behind `events.Recorder` was to decouple event generation from the side effect of firing the actual event, to improve testability and such. but I guess `record.FakeRecorder` can do that as well.